addons/project/models/project_task.py

```diff
@@ -167,8 +167,10 @@ def _read_group_personal_stage_type_ids(self, stages, domain, order):
         help="Date on which the state of your task has last been modified.\n"
             "Based on this information you can identify tasks that are stalling and get statistics on the time it usually takes to move tasks from one stage/state to another.")
 
-    project_id = fields.Many2one('project.project', string='Project', domain="['|', ('company_id', '=', False), ('company_id', '=?',  company_id)]", index=True, tracking=True, change_default=True)
-    display_in_project = fields.Boolean(default=True, readonly=True)
+    project_id = fields.Many2one('project.project', string='Project', domain="['|', ('company_id', '=', False), ('company_id', '=?',  company_id)]", inverse="_inverse_project_id", store=True, index=True, tracking=True, change_default=True)
+    display_in_project = fields.Boolean(compute='_compute_display_in_project', default=True, store=True, readonly=False)
+    # Technical field to display the 'Display in Project' button in the form view, depending on the project
+    show_display_in_project = fields.Boolean(compute='_compute_show_display_in_project', store=False)
```

Comment:
FYI, when we add a compute method, the field will be, by default, non-stored and readonly. 🙂 

```suggestion
    show_display_in_project = fields.Boolean(compute='_compute_show_display_in_project')
```